### ISSUE TITLE
Socket options SO_SNDBUF and SO_RCVBUF support

### ISSUE THREAD

**Issue by modatwork** [envoyproxy/gateway#2670]
*Description*:

Envoy will handle massive connections when used as L4 proxy. Setting the `SO_SNDBUF` and `SO_RCVBUF` socket options helps to limit the memory usage on each connection. So it would be nice to have these options in the APIs.

[optional *Relevant Links*:]

For downstream, Envoy has a `socket_options` in the listener config, https://www.envoyproxy.io/docs/envoy/v1.29.0/api-v3/config/listener/v3/listener.proto#config-listener-v3-listener

For upstream, Envoy has a `socket_options ` in the `upstream_bind_config` field of the cluster config, https://www.envoyproxy.io/docs/envoy/v1.29.0/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-cluster

**Comment by arkodg** [comment]
relates to https://github.com/envoyproxy/gateway/issues/2600

**Comment by aoledk** [comment]
How about adding socket options array to ClientTrafficPolicy and BackendTrafficPolicy, for setting socket options in xDS listener and xDS cluster, to limit kernel socket's memory consumption.

**Comment by zirain** [comment]
@arkodg is this fixed?

**Comment by aoledk** [comment]
> @arkodg is this fixed?

No, this feature's api is still not defined. If we **should** do, I can work on this.

**Comment by shawnh2** [comment]
reopen this one since lack of implementation
